tachyon/math/polynomials/univariate/radix2_evaluation_domain.h

```diff
@@ -73,6 +85,66 @@ class Radix2EvaluationDomain : public UnivariateEvaluationDomain<F, MaxDegree> {
     return base::bits::SafeLog2Ceiling(num_coeffs) <= F::Config::kTwoAdicity;
   }
 
+  void FFTBatch(RowMajorMatrix<F>& mat) {
+    CHECK(base::bits::IsPowerOfTwo(mat.rows()));
+    size_t log_h = this->log_size_of_group_;
+    mid_ = log_h / 2;
```

Comment:
I'm not sure what you are trying to do here. Maybe shouldn't you do  `CHECK_EQ(mat.rows(), this->size);`?